### PR TITLE
Refactor in-game sprite structs

### PR DESCRIPTION
**Pull request by IntelOrca**
In particular, change all sprite structs to inherit a common base and remove all the padding.

### CODE REVIEW

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -3526,7 +3526,7 @@ static void vehicle_update_collision_setup(rct_vehicle* vehicle)
     if (!(ride->lifecycle_flags & RIDE_LIFECYCLE_CRASHED))
     {
         rct_vehicle* frontVehicle = vehicle;
-        while (frontVehicle->is_child != 0)
+        while (frontVehicle->IsChild() != 0)
```
Comment (Gymnasiast) [inline src/openrct2/ride/Vehicle.cpp]:
```suggestion
        while (frontVehicle->IsChild())
```

**Comment by duncanspumpkin** [comment]
Should we drop `rct_` from the names or is that too big a change.

**Comment by Gymnasiast** [comment]
We definitely should, since they don't even match their RCT1/2 counterparts any more.

**Comment by IntelOrca** [comment]
Dropping the `rct_` will cause at least 5000 and in total almost 10000 changes across 388 files. Should we do that in it's own PR?

**Comment by Gymnasiast** [comment]
@IntelOrca Sure, that would make reviewing easier.

**Comment by ZehMatt** [comment]
It could be a general code maintenance PR where it also updates all our copyright dates since we already touch hundreds of files.

File: src/openrct2/ride/CableLift.cpp [hunk 2]
```diff
@@ -39,7 +39,7 @@ rct_vehicle* cable_lift_segment_create(
         move_sprite_to_list((rct_sprite*)current, SPRITE_LIST_TRAIN * 2);
         ride->cable_lift = current->sprite_index;
     }
-    current->is_child = head ? 0 : 1;
+    current->type = head ? 0 : 1;
```
Comment (Gymnasiast) [inline src/openrct2/ride/CableLift.cpp]: It would be better to use an enum for `type`.

Comment (IntelOrca) [reply src/openrct2/ride/CableLift.cpp]: How can I do that when type is different for each sprite identifier?

Comment (Gymnasiast) [reply src/openrct2/ride/CableLift.cpp]: I meant to say "define an enum with constants for 0 and 1, and leave type as an uint8_t".

Comment (IntelOrca) [reply src/openrct2/ride/CableLift.cpp]: done

**Pull request merged**